addons/website_form_project/controllers/main.py

```diff
@@ -36,3 +33,12 @@ def insert_record(self, request, model, values, custom, meta=None):
                 message_type='comment',
             )
         return res
+
+    def extract_data(self, model, values):
+        data = super().extract_data(model, values)
+        if model.model == 'project.task' and 'email_from' in values:
+            partner_id = request.env['res.partner'].sudo().search([('email', '=', values['email_from'])], limit=1).id
```

Comment:
Note that there are tools methods in mail to find a partner based on email, which notably tries to find normalized_emails, check for multiple emails, ... this search is a bit too naive :)

Reply:
Hello @tde-banana-odoo 
Necessary changes are done

Thank You :)